.github/workflows/build-agent-images.yml

```
@@ -39,6 +147,13 @@ jobs:
             runner: ubuntu-24.04-arm
             debarch: arm64
     runs-on: ${{ matrix.runner }}
+    # Skip matrix legs whose base did not change (detect gates them per base).
+    # A skipped leg is not a failure, so the release job still proceeds and
+    # keeps the unchanged base's existing published asset.
+    if: >-
```

actionlint (1.7.12)

[error] context "matrix" is not allowed here. available contexts are "github", "inputs", "needs", "vars". see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)